File: 77. Combinations/Solution.java
class Solution {
    public List<List<Integer>> combine(int n, int k) {
        List<List<Integer>> res = new ArrayList<>();
        if (k == 0) {
            res.add(new ArrayList<>());
            return res;
        }
        for (int i = k; i < n + 1; i++) {
            for (List<Integer> prev : combine(i - 1, k - 1)) {
                prev.add(i);
                res.add(prev);
            }
        }
        return res;
    }
}

File: 77. Combinations/Solution.py
class Solution:
    def combine(self, n: int, k: int) -> List[List[int]]:
        if k == 0: return [[]]
        res = []
        for i in range(k, n + 1):
            for prev in self.combine(i - 1, k - 1):
                res.append(prev + [i])
        return res
